### DISCUSSION TITLE
[CRASH] Docker service crashes when browsing a large number of images(about 17000+)

### DISCUSSION THREAD

**Discussion by BBBOND** [photoprism/photoprism#3989] [Q&A] [answered]
PLEASE PROCEED ONLY IF YOU ARE ABSOLUTELY SURE THAT THIS IS NOT A TECHNICAL SUPPORT INCIDENT AND/OR POSSIBLY A PROBLEM WITH SOME OTHER SOFTWARE YOU ARE USING. VISIT <https://www.photoprism.app/kb/getting-support> TO LEARN MORE ABOUT OUR SUPPORT OPTIONS. THANK YOU FOR YOUR CAREFUL CONSIDERATION!

---------------------------------------------------------------------------

#### 1. What is not working as documented?

Docker service crashes when browsing a large number of images(about 17000+)

#### 2. How can we reproduce it?

1. Import 17000+ pictures
2. Open "http://xxx.com/library/browse"
3. Scroll down a few pages

When reporting an import, indexing, or performance issue, please include the number and type of pictures in your library, as well as any configuration options you have changed, such as for thumbnail quality.

#### 3. What behavior do you expect?

Don't crash

#### 4. What could be the cause of your problem?

```
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1650 +0x1 fp=0xc0033b07e8 sp=0xc0033b07e0 pc=0x472e61
created by net/http.(*connReader).startBackgroundRead in goroutine 468
        /usr/local/go/src/net/http/server.go:679 +0xba

goroutine 473 [runnable]:
net/http.(*connReader).startBackgroundRead.func2()
        /usr/local/go/src/net/http/server.go:679 fp=0xc001b4a7e0 sp=0xc001b4a7d8 pc=0x7a2b00
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1650 +0x1 fp=0xc001b4a7e8 sp=0xc001b4a7e0 pc=0x472e61
created by net/http.(*connReader).startBackgroundRead in goroutine 451
        /usr/local/go/src/net/http/server.go:679 +0xba

goroutine 459 [IO wait]:
runtime.gopark(0xc0034a1a70?, 0xb?, 0x0?, 0x0?, 0x40?)
        /usr/local/go/src/runtime/proc.go:398 +0xce fp=0xc003481da0 sp=0xc003481d80 pc=0x442e2e
runtime.netpollblock(0x488218?, 0x40cd66?, 0x0?)
        /usr/local/go/src/runtime/netpoll.go:564 +0xf7 fp=0xc003481dd8 sp=0xc003481da0 pc=0x43b817
internal/poll.runtime_pollWait(0x150e90b24820, 0x72)
        /usr/local/go/src/runtime/netpoll.go:343 +0x85 fp=0xc003481df8 sp=0xc003481dd8 pc=0x46d405
internal/poll.(*pollDesc).wait(0xc0000fbf00?, 0xc0035b8ca1?, 0x0)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27 fp=0xc003481e20 sp=0xc003481df8 pc=0x4adb27
internal/poll.(*pollDesc).waitRead(...)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0xc0000fbf00, {0xc0035b8ca1, 0x1, 0x1})
        /usr/local/go/src/internal/poll/fd_unix.go:164 +0x27a fp=0xc003481eb8 sp=0xc003481e20 pc=0x4aee1a
net.(*netFD).Read(0xc0000fbf00, {0xc0035b8ca1?, 0x0?, 0x0?})
        /usr/local/go/src/net/fd_posix.go:55 +0x25 fp=0xc003481f00 sp=0xc003481eb8 pc=0x69bf65
net.(*conn).Read(0xc004bd7a18, {0xc0035b8ca1?, 0x0?, 0x0?})
        /usr/local/go/src/net/net.go:179 +0x45 fp=0xc003481f48 sp=0xc003481f00 pc=0x6aa205
net.(*TCPConn).Read(0x0?, {0xc0035b8ca1?, 0x0?, 0x0?})
        <autogenerated>:1 +0x25 fp=0xc003481f78 sp=0xc003481f48 pc=0x6bcbc5
net/http.(*connReader).backgroundRead(0xc0035b8c90)
        /usr/local/go/src/net/http/server.go:683 +0x37 fp=0xc003481fc8 sp=0xc003481f78 pc=0x7a2bf7
net/http.(*connReader).startBackgroundRead.func2()
        /usr/local/go/src/net/http/server.go:679 +0x25 fp=0xc003481fe0 sp=0xc003481fc8 pc=0x7a2b25
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1650 +0x1 fp=0xc003481fe8 sp=0xc003481fe0 pc=0x472e61
created by net/http.(*connReader).startBackgroundRead in goroutine 453
        /usr/local/go/src/net/http/server.go:679 +0xba

goroutine 411 [IO wait]:
runtime.gopark(0xc0033b39e0?, 0xb?, 0x0?, 0x0?, 0x3d?)
        /usr/local/go/src/runtime/proc.go:398 +0xce fp=0xc0033b0da0 sp=0xc0033b0d80 pc=0x442e2e
runtime.netpollblock(0x488218?, 0x40cd66?, 0x0?)
        /usr/local/go/src/runtime/netpoll.go:564 +0xf7 fp=0xc0033b0dd8 sp=0xc0033b0da0 pc=0x43b817
internal/poll.runtime_pollWait(0x150e90bd7978, 0x72)
        /usr/local/go/src/runtime/netpoll.go:343 +0x85 fp=0xc0033b0df8 sp=0xc0033b0dd8 pc=0x46d405
internal/poll.(*pollDesc).wait(0xc0020af700?, 0xc0035b8c41?, 0x0)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27 fp=0xc0033b0e20 sp=0xc0033b0df8 pc=0x4adb27
internal/poll.(*pollDesc).waitRead(...)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0xc0020af700, {0xc0035b8c41, 0x1, 0x1})
        /usr/local/go/src/internal/poll/fd_unix.go:164 +0x27a fp=0xc0033b0eb8 sp=0xc0033b0e20 pc=0x4aee1a
net.(*netFD).Read(0xc0020af700, {0xc0035b8c41?, 0xc0033b3b50?, 0xc0033b3b60?})
        /usr/local/go/src/net/fd_posix.go:55 +0x25 fp=0xc0033b0f00 sp=0xc0033b0eb8 pc=0x69bf65
net.(*conn).Read(0xc0018c4270, {0xc0035b8c41?, 0xc004eda1f8?, 0xc004eda224?})
        /usr/local/go/src/net/net.go:179 +0x45 fp=0xc0033b0f48 sp=0xc0033b0f00 pc=0x6aa205
net.(*TCPConn).Read(0xc004eda227?, {0xc0035b8c41?, 0xc004eda230?, 0xc004eda250?})
        <autogenerated>:1 +0x25 fp=0xc0033b0f78 sp=0xc0033b0f48 pc=0x6bcbc5
net/http.(*connReader).backgroundRead(0xc0035b8c30)
        /usr/local/go/src/net/http/server.go:683 +0x37 fp=0xc0033b0fc8 sp=0xc0033b0f78 pc=0x7a2bf7
net/http.(*connReader).startBackgroundRead.func2()
        /usr/local/go/src/net/http/server.go:679 +0x25 fp=0xc0033b0fe0 sp=0xc0033b0fc8 pc=0x7a2b25
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1650 +0x1 fp=0xc0033b0fe8 sp=0xc0033b0fe0 pc=0x472e61
created by net/http.(*connReader).startBackgroundRead in goroutine 432
        /usr/local/go/src/net/http/server.go:679 +0xba

rax    0x0
rbx    0x15
rcx    0x150edac9999b
rdx    0x6
rdi    0x14
rsi    0x15
rbp    0x6
rsp    0x150e923e9710
r8     0x0
r9     0x0
r10    0x8
r11    0x246
r12    0x150e923e99e0
r13    0x16
r14    0xc0000069c0
r15    0x11
rip    0x150edac9999b
rflags 0x246
cs     0x33
fs     0x0
gs     0x0
```

#### 5. Can you provide us with example files for testing, error logs, or screenshots?

Sorry I can't provide example files, those are private photos.

#### 6. Which software versions do you use?

(a) PhotoPrism Architecture & Build Number: AMD64, Build 231128-f48ff16ef

(b) Database Type & Version: SQLite

(c) Operating System Types & Versions: Linux

(d) Browser Types & Versions: Chrome

(e) Ad Blockers, Browser Plugins, and/or Firewall Software? No

#### 7. On what kind of device is PhotoPrism installed?

(a) Device / Processor Type: Intel Core i5-9600

(b) Physical Memory & Swap Space in GB: 20GB

(c) Storage Type: HDD and with SSD cache

(d) Anything else that might be helpful to know?

docker is run on Unraid OS.

#### 8. Do you use a Reverse Proxy, Firewall, VPN, or CDN?
Use "Nginx Proxy Manager" in docker.

**Comment by graciousgrey** [comment]
As no one else is having this problem I would assume that it is most likely caused by your set up.

Can you please confirm that your server meets the minimum system requirements:

https://docs.photoprism.app/getting-started/#system-requirements

Do you have at least 4GB SWAP configured? Missing SWAP has been the most common reason for server crashes.

Also please note that [SQLite is not a good choice](https://docs.photoprism.app/getting-started/troubleshooting/sqlite/) for users who require scalability and high performance. If it is used in a configuration template, we recommend changing it to use MariaDB instead.

**Comment by Fireflaker** [comment, marked as answer]
I have over 10 times that (approx 190000 items) and no crashes. Running on 16 GB ram, 4 core intel CPU, 15 doller gpu, truenas scale.

Consider checking if the physical drives are running healthy or getting enough power. Not sure about your system, but truenas apparently has no countermeasure toward an underpowered 5v rail, to the extent of kernel panic hard crash restart whenever a corrupted pool is mounted. What that means is there may be unhandled data corruption that has no warning, no error, no log, and cannot be fixed with scrubbing/resilvering. and there is no way around it until you find the exact sector of corrupted data manually. (and you still cannot delete it, you must copy all good parts to a new pool, erase the old, and copy it all back).

You can do that by smb sharing everything, and going through the originals to see if it is a photoprism problem or something more fundamental.

**Reply by BBBOND** [reply]
thanks for the test, I replaced to MariaDB, and never crash again.
